domain/relation/service/relation.go

```diff
@@ -412,15 +433,15 @@ func (s *Service) GetRelationUUIDByKey(ctx context.Context, relationKey corerela
 	case 2:
 		uuid, err = s.st.GetRegularRelationUUIDByEndpointIdentifiers(
 			ctx,
-			endpointIdentifiers[0],
-			endpointIdentifiers[1],
+			eids[0],
+			eids[1],
 		)
 		if err != nil {
 			return "", errors.Errorf("getting regular relation by key: %w", err)
 		}
 		return uuid, nil
 	default:
-		return "", errors.Errorf("expected 1 or 2 endpoints in relation key, got %d", len(endpointIdentifiers))
+		return "", errors.Errorf("expected 1 or 2 endpoints in relation key, got %d", len(eids))
```

Comment:
suggestion: now that `relationKey.Validate()` is being called above, this case should never happen, thus a programming error. Update the message to say that.

Reply:
Prepended with `internal error:` and simplified error message.